filament/src/components/RenderableManager.cpp

```diff
@@ -204,83 +207,30 @@ RenderableManager::Builder& RenderableManager::Builder::enableSkinningBuffers(bo
 }
 
 RenderableManager::Builder& RenderableManager::Builder::boneIndicesAndWeights(size_t primitiveIndex,
-               math::float2 const* indicesAndWeights, size_t count) {
-    std::vector<Entry>& entries = mImpl->mEntries;
-
-    ASSERT_PRECONDITION(primitiveIndex < entries.size() && primitiveIndex >= 0,
-        "[primitive @ %u] index is out of size (%u)", primitiveIndex, entries.size());
-    auto vertexCount = entries[primitiveIndex].vertices->getVertexCount();
-    ASSERT_PRECONDITION(vertexCount > 0,
-        "[primitive @ %u] no defined vertices (%u)", primitiveIndex, vertexCount);
-    auto pairsPerVertexCount = count / vertexCount;
-    ASSERT_PRECONDITION(pairsPerVertexCount > 0 && count % vertexCount == 0,
-        "[primitive @ %u] bone indices and weights pairs count (%u) must be a multiple of vertex count (%u)",
-        primitiveIndex, count, vertexCount);
-
+               math::float2 const* indicesAndWeights, size_t count, size_t bonesPerVertex) noexcept{
+    auto vertexCount = count / bonesPerVertex;
     utils::FixedCapacityVector<utils::FixedCapacityVector<filament::math::float2>> bonePairs(vertexCount);
     for( size_t iVertex = 0; iVertex < vertexCount; iVertex++){
-        utils::FixedCapacityVector<float2> vertexData(pairsPerVertexCount);
-        std::memcpy(&vertexData[0], &indicesAndWeights[iVertex * pairsPerVertexCount],
-             pairsPerVertexCount * sizeof(float2));
+        utils::FixedCapacityVector<float2> vertexData(bonesPerVertex);
+        std::memcpy(&vertexData[0], &indicesAndWeights[iVertex * bonesPerVertex],
+             bonesPerVertex * sizeof(float2));
```

Comment:
use `std::copy_n` instead of `memcpy`, at the very least use `vertexData.data()` instead of `&[0]`